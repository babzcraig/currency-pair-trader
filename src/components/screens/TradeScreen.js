import React from "react";
import PropTypes from "prop-types";

const TradeScreen = ({ amountToBuy, onInputChange }) => {
  return (
    <React.Fragment>
      <h2 className="trade-section">Trade</h2>
      <div>
        <input type="number" placeholder="USD" disabled />
      </div>
      <div>
        <input
          className="amount-to-buy"
          type="text"
          placeholder="Enter your amount"
          onChange={onInputChange}
          value={amountToBuy}
        />
      </div>
    </React.Fragment>
  );
};

TradeScreen.propTypes = {
  amountToBuy: PropTypes.string,
  onInputChange: PropTypes.func
};

export default TradeScreen;
