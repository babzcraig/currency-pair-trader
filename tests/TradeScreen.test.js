import React from "react";
import ReactDOM from "react-dom";
import TradeScreen from "../src/components/screens/TradeScreen";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<TradeScreen />", () => {
  const wrapper = shallow(<TradeScreen />);

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TradeScreen />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("contains two input fields", () => {
    expect(wrapper.find("input").length).toBe(2);
  });

  it('contains an h2 with a Text of "Trade"', () => {
    expect(wrapper.find("h2").text()).toBe("Trade");
  });

  it('contains an h2 with a className of "trade-section', () => {
    expect(wrapper.find("h2").hasClass("trade-section")).toBe(true);
  });

  it("matches the snapshot", () => {
    const tree = shallow(<TradeScreen />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
