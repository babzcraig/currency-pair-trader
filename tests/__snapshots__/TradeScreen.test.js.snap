// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradeScreen /> matches the snapshot 1`] = `
<React.Fragment>
  <h2
    className="trade-section"
  >
    Trade
  </h2>
  <div>
    <input
      disabled={true}
      placeholder="USD"
      type="number"
    />
  </div>
  <div>
    <input
      className="amount-to-buy"
      placeholder="Enter your amount"
      type="text"
    />
  </div>
</React.Fragment>
`;
